src/transactions/TransactionFrame.cpp

```diff
@@ -972,6 +993,16 @@ TransactionFrame::checkValidWithOptionallyChargedFee(
                     upperBoundCloseTimeOffset) == ValidationType::kMaybeValid;
     if (res)
     {
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+        // Check that smart ops are not mixed with classic ops. Eventually we
+        // should separately classify 'smart' transactions, but for now a simple
+        // check should be sufficient.
```

Comment:
should this be checked in `commonValid` before signature verification?

Reply:
Good point. Moved.